flink-metrics/flink-metrics-prometheus/src/test/java/org/apache/flink/metrics/prometheus/PrometheusPushGatewayReporterTest.java

```diff
@@ -49,4 +55,52 @@ public void testParseIncompleteGroupingKey() {
         groupingKey = PrometheusPushGatewayReporterFactory.parseGroupingKey("k1");
         Assert.assertTrue(groupingKey.isEmpty());
     }
+
+    @Test
+    public void testConnectToPushGatewayUsingHostAndPort() {
+        PrometheusPushGatewayReporterFactory factory = new PrometheusPushGatewayReporterFactory();
+        MetricConfig metricConfig = new MetricConfig();
+        metricConfig.setProperty(HOST.key(), "localhost");
+        metricConfig.setProperty(PORT.key(), "18080");
+        PrometheusPushGatewayReporter reporter = factory.createMetricReporter(metricConfig);
+        String gatewayBaseURL =
+                (String) Whitebox.getInternalState(reporter.getPushGateway(), "gatewayBaseURL");
+        Assert.assertEquals(gatewayBaseURL, "http://localhost:18080/metrics/");
+    }
+
+    @Test
+    public void testConnectToPushGatewayUsingHostUrl() {
+        PrometheusPushGatewayReporterFactory factory = new PrometheusPushGatewayReporterFactory();
+        MetricConfig metricConfig = new MetricConfig();
+        metricConfig.setProperty(HOST_URL.key(), "https://localhost:18080");
+        PrometheusPushGatewayReporter reporter = factory.createMetricReporter(metricConfig);
+        String gatewayBaseURL =
+                (String) Whitebox.getInternalState(reporter.getPushGateway(), "gatewayBaseURL");
+        Assert.assertEquals(gatewayBaseURL, "https://localhost:18080/metrics/");
+    }
+
+    @Test
+    public void testConnectToPushGatewayPreferHostUrl() {
+        PrometheusPushGatewayReporterFactory factory = new PrometheusPushGatewayReporterFactory();
+        MetricConfig metricConfig = new MetricConfig();
+        metricConfig.setProperty(HOST_URL.key(), "https://localhost:18080");
+        metricConfig.setProperty(HOST.key(), "localhost1");
+        metricConfig.setProperty(PORT.key(), "18081");
+        PrometheusPushGatewayReporter reporter = factory.createMetricReporter(metricConfig);
+        String gatewayBaseURL =
+                (String) Whitebox.getInternalState(reporter.getPushGateway(), "gatewayBaseURL");
```

Comment:
Please find a ways to test this without using mockito, e.g., by storing the URL in the reporter and making it accessible, or moving the host/port handling into a static `@VisibleForTesting` method similar to `parseGroupingKey`.